iup/__init__.py

```diff
@@ -1088,186 +1088,64 @@ def check_date_match(data: IncidentUptakeData, pred: PointForecast):
     ), "Duplicated dates are found in data and prediction."
 
 
-def calculate_mspe(data: IncidentUptakeData, pred: PointForecast) -> pl.DataFrame:
+def score(
+    data: IncidentUptakeData,
+    pred: PointForecast,
+    score_fun: Callable[[pl.Expr, pl.Expr], pl.Expr],
+) -> pl.DataFrame:
     """
-    Calculate MSPE between observed data and forecast.
+    Calculate score between observed data and forecast.
     ----------------------
 
     Parameters
     data:
         The observed data used for modeling. Should be IncidentUptakeData
     pred:
         The forecast made by model. Should be PointForecast
+    score_fun:
+        Scoring function. Takes observed and true values.
 
     Return
-    pl.DataFrame with forecast start date, forecast end date, and MSPE
+    pl.DataFrame with one row: forecast start date, forecast end date, and score
 
     """
-    joined = data.join(pred, on="date", how="inner", validate="1:1")
+    # validate inputs
+    data = IncidentUptakeData(data)
```

Comment:
Our evaluation metrics assume that the data and predictions are incident (not cumulative) uptake. So here, should we _check_ that `data` is `IncidentUptakeData` rather than _coercing_ it into `IncidentUptakeData`? Seems like one could accidentally provide `CumulativeUptakeData` (which passes all the `IncidentUptakeData` validation, I believe) and get a (wrong) answer with no error.

Reply:
fair, maybe this should be `isinstance(data, IncidentUptakeData)`?

Reply:
Yeah, I think that would be a good fix.

Reply:
Is the validation already done in the arguments? Like the score function explicitly asks for `IncidenUptakeData` as the type of data.